### PR TITLE
Fix documentation: using numerical target ports in services doesn't cause downtime

### LINKED ISSUE #1650: Downtime during migration to V2
I am trying to upgrade alb ingress controller v1.1.8 to new LB controller v2.0.0, as per this doc -> https://kubernetes-sigs.github.io/aws-load-balancer-controller/guide/upgrade/migrate_v1_v2/

I tried upgrading in my dev cluster and we saw that we are facing downtime during the upgrade process.

Result of the test during the upgrade process :point_down:

Test-1:
```
fortio load -a -c 50 -qps 500 -t 300s http://xxx-2048game-2048ingr-6fa0-xxx.ap-south-1.elb.amazonaws.com 
Code 200 : 148983 (99.3 %)
Code 503 : 1017 (0.7 %)
```
Test-2:
```
fortio load -a -c 50 -qps 500 -t 200s http://xxx-2048game-2048ingr-6fa0-xxx.ap-south-1.elb.amazonaws.com 
Code 200 : 87031 (87.0 %)
Code 503 : 12969 (13.0 %)
```

Currently, during the upgrade process before the targets are registered to the new target group, the target group is updated in ALB listeners, so where there are 0 targets in the new target group the ALB listener will be pointing to that target group, that is causing the downtime.
This shouldn’t be the case  ALB listener should be updated only once the targets are registered to the new target group.

### PR DESCRIPTION
**Pull request by devkid**
The actual issue was fixed in #1635.

Fixes #1650.

### CODE REVIEW

File: docs/guide/upgrade/migrate_v1_v2.md [hunk 1]
```diff
@@ -15,17 +15,11 @@ It supports existing AWS resources provisioned by AWSALBIngressController(>=v1.1
...
-    !!!warning "downtimes"
-        This would cause downtimes to your service during targets registration into new TargetGroups created.
-    
-    !!!tip "details"
-        * The AWSALBIngressController always used `1` as TargetGroup's port.
-        * The AWSLoadBalancerController will use 
-            * the actual numeric TargetPort as TargetGroup's port if a numeric TargetPort used.
-            * `1` as TargetGroup's port if a lexical TargetPort used.
-        * The AWSLoadBalancerController will automatically create new TargetGroups and cleanup old TargetGroups if any.
+2. The AWSALBIngressController always used `1` as TargetGroup's port. The AWSLoadBalancerController will use
```
Comment (kishorj) [inline docs/guide/upgrade/migrate_v1_v2.md:18]: the !!!tip annotation is for mkdocs rendering - https://kubernetes-sigs.github.io/aws-load-balancer-controller/latest/guide/upgrade/migrate_v1_v2/
Please keep mkdocs the annotation. 

You can also verify the livedocs rendering on your workspace - 
```
pip3 install mike
mike deploy v2.0
mike serve
```
mike is a mkdocs plugin for versioned docs https://github.com/jimporter/mike

File: docs/guide/upgrade/migrate_v1_v2.md [hunk 2]
```diff
@@ -15,17 +15,11 @@ It supports existing AWS resources provisioned by AWSALBIngressController(>=v1.1
...
-        * The AWSALBIngressController always used `1` as TargetGroup's port.
-        * The AWSLoadBalancerController will use 
-            * the actual numeric TargetPort as TargetGroup's port if a numeric TargetPort used.
-            * `1` as TargetGroup's port if a lexical TargetPort used.
-        * The AWSLoadBalancerController will automatically create new TargetGroups and cleanup old TargetGroups if any.
+2. The AWSALBIngressController always used `1` as TargetGroup's port. The AWSLoadBalancerController will use 
+   * the actual numeric `targetPort` of the corresponding Kubernetes `Service` as TargetGroup's port if a numeric `targetPort` is used.
+   * `1` as TargetGroup's port if a lexical `targetPort` is used.
+   
+   Existing TargetGroups will be preserved; if a numerical `targetPort` is used, the TargetGroup's port will not be updated and stay `1`.
```
Comment (kishorj) [inline docs/guide/upgrade/migrate_v1_v2.md:22]: You can put this line in the description for this list item 
```
2. Existing Targetgroups ...

      !!! tip ...
```

**Review by kishorj — COMMENTED** [review]
Thanks for your changes, I have some minor comments, appreciate if you can address them.

File: docs/guide/upgrade/migrate_v1_v2.md (hunk 2 above)
Comment (devkid) [reply docs/guide/upgrade/migrate_v1_v2.md]: I moved the
```
Existing TargetGroups will be preserved; if a numerical `targetPort` is used, the TargetGroup's port will not be updated and stay `1`
```
into a "details" block.

Comment (devkid) [reply docs/guide/upgrade/migrate_v1_v2.md]: Let me know if that's what you want.

File: docs/guide/upgrade/migrate_v1_v2.md [hunk 3]
```diff
@@ -15,17 +15,12 @@ It supports existing AWS resources provisioned by AWSALBIngressController(>=v1.1
...
+     * the actual numeric `targetPort` of the corresponding Kubernetes `Service` as TargetGroup's port if a numeric `targetPort` is used.
+     * `1` as TargetGroup's port if a lexical `targetPort` is used.
+   
     !!!tip "details"
-        * The AWSALBIngressController always used `1` as TargetGroup's port.
-        * The AWSLoadBalancerController will use 
-            * the actual numeric TargetPort as TargetGroup's port if a numeric TargetPort used.
-            * `1` as TargetGroup's port if a lexical TargetPort used.
-        * The AWSLoadBalancerController will automatically create new TargetGroups and cleanup old TargetGroups if any.
+        Existing TargetGroups will be preserved; if a numerical `targetPort` is used, the TargetGroup's port will not be updated and stay `1`.
```
Comment (kishorj) [inline docs/guide/upgrade/migrate_v1_v2.md:23]: if numerical `targetPort` is used, then the port will be updated

Comment (kishorj) [reply docs/guide/upgrade/migrate_v1_v2.md:23]: This was a bit confusing, I got what you were trying to say.

Comment (devkid) [reply docs/guide/upgrade/migrate_v1_v2.md:23]: So should I remove the whole second point in that list?

Comment (M00nF1sh) [reply docs/guide/upgrade/migrate_v1_v2.md:23]: @devkid i think we should complete remove it. 
better to mention somewhere in the doc that this doc applies to v2.0.1.

Comment (kishorj) [reply docs/guide/upgrade/migrate_v1_v2.md:23]: I prefer removing the second point completely.

Comment (devkid) [reply docs/guide/upgrade/migrate_v1_v2.md]: Second point is removed, version in the paragraph before the first point is updated to 2.0.1.

Comment (kishorj) [reply docs/guide/upgrade/migrate_v1_v2.md]: Thanks.

**Review by kishorj — APPROVED** [review]

**Pull request merged**